### ISSUE TITLE
[Bug] The keys related to `gr` can not removed during lsp attach

### ISSUE THREAD

**Issue by chaozwn** [AstroNvim/AstroNvim#2665]
### Checklist

- [X] I have searched through the AstroNvim docs
- [X] I have searched through the existing issues of AstroNvim
- [X] I have searched the existing issues of plugins related to this issue
- [X] I have run `:Lazy update` and are on the latest version of AstroNvim
- [ ] I can replicate the bug with the minimal `repro.lua` provided below

### Operating system/version

macos 14.5

### Terminal/GUI

Wezterm

### AstroNvim Health

AstroNvim ~
- AstroNvim Version: v4.22.0-dev-1-gdaf0b60
- Neovim Version: v0.10.0
- WARNING Neovim nightly is not officially supported and may have breaking changes
- OK `git` is installed: Used for core functionality such as updater and plugin management
- OK `open` is installed: Used for `gx` mapping for opening files with system opener (Optional)
- OK `rg` is installed: Used for Telescope `live_grep` picker, `<Leader>fw` and `<Leader>fW` by default (Optional)
- OK `lazygit` is installed: Used for mappings to pull up git TUI (Optional)
- OK `node` is installed: Used for mappings to pull up node REPL (Optional)
- OK `gdu-go` is installed: Used for mappings to pull up disk usage analyzer (Optional)
- OK `btm` is installed: Used for mappings to pull up system monitor (Optional)
- OK `python` is installed: Used for mappings to pull up python REPL (Optional)

### Describe the bug

You know telescope's lsp functionality is not complete, but I don't like the default functionality. 
So I set it to automatically remove the default keymap set by astronvim and use telescope instead,  but it doesn't seem to remove success.

![image](https://github.com/user-attachments/assets/ece00214-31c8-4d90-bf60-4bf7690ad7c7)

### Steps to Reproduce

my lua config
```lua
local utils = require "astrocore"

---@type LazySpec
return {
  {
    "AstroNvim/astrolsp",
    ---@type AstroLSPOpts
    opts = {
      ---@diagnostic disable: missing-fields
      config = {
        lua_ls = {
          on_attach = function()
            vim.keymap.del("n", "grr", { buffer = true })
            vim.keymap.del("n", "grn", { buffer = true })
            vim.keymap.del("n", "gra", { buffer = true })
            vim.keymap.del("x", "gra", { buffer = true })
          end,
          settings = { Lua = { hint = { enable = true, arrayIndex = "Disable" } } },
        },
      },
    },
  },
  {
    "nvim-treesitter/nvim-treesitter",
    optional = true,
    opts = function(_, opts)
      if opts.ensure_installed ~= "all" then
        opts.ensure_installed = utils.list_insert_unique(opts.ensure_installed, { "lua", "luap" })
      end
    end,
  },
  {
    "williamboman/mason-lspconfig.nvim",
    optional = true,
    opts = function(_, opts) opts.ensure_installed = utils.list_insert_unique(opts.ensure_installed, { "lua_ls" }) end,
  },
  {
    "jay-babu/mason-null-ls.nvim",
    optional = true,
    opts = function(_, opts)
      opts.ensure_installed = require("astrocore").list_insert_unique(opts.ensure_installed, { "stylua", "selene" })
    end,
  },
}
```

### Expected behavior

Able to remove keys correctly

### Repro

```Lua
-- save as repro.lua
-- run with nvim -u repro.lua
-- DO NOT change the paths
local root = vim.fn.fnamemodify("./.repro", ":p")

-- set stdpaths to use .repro
for _, name in ipairs({ "config", "data", "state", "runtime", "cache" }) do
  vim.env[("XDG_%s_HOME"):format(name:upper())] = root .. "/" .. name
end

-- bootstrap lazy
local lazypath = root .. "/plugins/lazy.nvim"
if not vim.loop.fs_stat(lazypath) then
  -- stylua: ignore
  vim.fn.system({ "git", "clone", "--filter=blob:none", "https://github.com/folke/lazy.nvim.git", "--branch=stable", lazypath })
end
vim.opt.rtp:prepend(vim.env.LAZY or lazypath)

-- install plugins
local plugins = {
  { "AstroNvim/AstroNvim", import = "astronvim.plugins" },

  -- add any other plugins/customizations here
}
require("lazy").setup(plugins, {
  root = root .. "/plugins",
})

-- add anything else here (autocommands, vim.filetype, etc.)
```

**Comment by mehalter** [comment]
First, this is an invalid issue given that the minimal `repro.lua` is not properly built to reproduce the bug. Please make sure you have done this correctly as you indicated in the checklist.

Second, this is not a bug. You are getting an error that says that it cannot delete the mapping because it doesn't exist. Language servers only set up mappings that are supported by the language server.

**Comment by mehalter** [comment]
Here is a place in the docs where we show the correct way to do this: https://docs.astronvim.com/recipes/mappings/#enable-telescope-lsp-mappings

**Comment by chaozwn** [comment]
> https://docs.astronvim.com/recipes/mappings/#enable-telescope-lsp-mappings

The effect of this document is not good. My ideal effect is that according to different LSPS, if telescope support this lsp, we will add this mapping, if not, we will use your setting mapping.

**Comment by mehalter** [comment]
Ok, then you should add proper error handling and only delete mappings that exist

**Comment by chaozwn** [comment]
> Ok, then you should add proper error handling and only delete mappings that exist

Actually, I did, but you put the mapping of these LSP keymapping like `gr` in the astrocore module, which gave me a headache. I couldn't delete him, or rather, there was no right place to delete him. Whether the `grn`,`gra`,`grr` can be moved to `astrolsp`.

**Comment by mehalter** [comment]
It can, you can simply move them in your user configuration and set them up however you want

**Comment by chaozwn** [comment]
> It can, you can simply move them in your user configuration and set them up however you want

I can see your point, but are these keymappings really appropriate in this file?

<img width="961" alt="image" src="https://github.com/user-attachments/assets/6dc20acf-3a61-4580-91d0-0e7d9f5a0f61">

**Comment by mehalter** [comment]
This is a backport of the default mapping in neovim that will be released in v0.11. So I suppose it's good we are getting used to them now before we have no choice at all 😅

**Comment by chaozwn** [comment]
> This is a backport of the default mapping in neovim that will be released in v0.11. So I suppose it's good we are getting used to them now before we have no choice at all 😅

You may misunderstand me, I think this code block should be moved to 'astrolsp_mapping.lua'.

**Comment by mehalter** [comment]
Yeah, I do actually agree. But this is meant to reflect the default mappings in Neovim 0.11 where those mappings are core and set by default if there is a language server attached or not. They are fully global. We are reflecting that mapping setup now together get used to it and prepare to remove those mappings from AstroNvim all together down the line when we drop support from v0.10
